### PR TITLE
[2.8.x]: Fix release process

### PR DESCRIPTION
**Pull request by marcospereira**
Two major problems:

1. It was not promoting the staged repository
2. It was generating multiple staging repositories

This should fix both by using the defaults provided by interplay.

### CODE REVIEW

**Comment by renatocaval** [comment]
@marcospereira, something went wrong in Travis. Build is failing with error in build definition.

**Comment by ignasi35** [comment]
> @marcospereira, something went wrong in Travis. Build is failing with error in build definition.

```
Failed to sort List(
  interplay.PlayNoPublishBase, 
  org.scalajs.sbtplugin.ScalaJSPlugin, 
  interplay.PlayRootProjectBase, 
  sbt.plugins.CorePlugin, 
  interplay.PlaySonatypeBase, 
  sbt.plugins.JvmPlugin, 
  interplay.PlayBuildBase, 
  interplay.PlayReleaseBase, 
  sbt.plugins.IvyPlugin, 
  sbt.plugins.Giter8TemplatePlugin, 
  scalajscrossproject.ScalaJSCrossPlugin, 
  interplay.PlayBintrayBase, 
  de.heikoseeberger.sbtheader.HeaderPlugin, 
  org.scalafmt.sbt.ScalafmtPlugin, 
  org.scalajs.sbtplugin.internal.ScalaJSGlobalPlugin, 
  sbtwhitesource.WhiteSourcePlugin, 
  com.typesafe.tools.mima.plugin.MimaPlugin, 
  org.portablescala.sbtplatformdeps.PlatformDepsPlugin, 
  sbt.plugins.SemanticdbPlugin, 
  sbt.plugins.JUnitXmlReportPlugin, 
  sbtcrossproject.CrossPlugin, 
  com.jsuereth.sbtpgp.SbtPgp, 
  sbtrelease.ReleasePlugin, 
  interplay.PlayWhitesourcePlugin) 
topologically
```
Sounds like a [cyclical dependency on the plugins](https://travis-ci.com/playframework/play-json/jobs/258409826#L270).

File: build.sbt [hunk 1]
```diff
@@ -132,18 +126,15 @@ lazy val commonSettings = Def.settings(
 
 lazy val root = project
   .in(file("."))
-  .enablePlugins(PlayRootProject, ScalaJSPlugin)
+  .enablePlugins(PlayRootProject, ScalaJSPlugin, PlayNoPublish)
```
Comment (ignasi35) [inline build.sbt]: (I'll ignore `ScalaJSPlugin`)

The `enablePlugins` above can be expanded to:

1. `PlayRootProject` expands to `PlayRootProjectBase &&! PlaySbtCompat.optScriptedAutoPlugin`
1. `PlayNoPublish` expands to `PlayNoPublishBase &&! BintrayPlugin &&! Sonatype`

Then we have:

```
enablePlugins(
  PlayRootProjectBase &&! PlaySbtCompat.optScriptedAutoPlugin, // exclude(ScriptedAutoPlugin)
  PlayNoPublishBase &&! BintrayPlugin &&! Sonatype // exclude(BintrayPlugin, Sonatype)
)
```

Now, `PlayRootProjectBase` brings some stuff transitively:

```
object PlayRootProjectBase extends AutoPlugin {
  override def trigger = noTrigger
  override def requires = PlayBuildBase && PlayBintrayBase && PlaySonatypeBase && PlayReleaseBase
```

So the end result looks like:

```
enablePlugins(
  PlayRootProjectBase &&! PlaySbtCompat.optScriptedAutoPlugin, // exclude(ScriptedAutoPlugin)
  PlayBuildBase, 
  PlayBintrayBase,  // brings `BintrayPlugin` transitively
  PlaySonatypeBase, // brings `Sonatype` transitively
  PlayReleaseBase,

  PlayNoPublishBase &&! BintrayPlugin &&! Sonatype // exclude(BintrayPlugin, Sonatype)
  PlayBuildBase,    // transitively added by `PlayNoPublishBase`
)
```

Which, I guess, looks like:

```
enablePlugins(
  PlayRootProjectBase
  PlayBuildBase, 
  PlayReleaseBase,
  PlayNoPublishBase
)
```

- - - - - - - - - - - 

IIUC, the purpose is to not publish the `root` project. In that case, we don't want `PlayReleaseBase`. 

Then, we could be tempted to also remove `PlayRootProjectBase` (which is the one bringing in `PlayReleaseBase` but `PlayRootProjectBase` is defining the [`crossScalaVersions`](https://github.com/playframework/interplay/blob/master/src/main/scala/interplay/PlayBuildBase.scala#L356-L362))

I don't know what I'm talking about anymore.

 - - - - - - - - - - 

OMG, this is so confusing.

According to [the docs](https://github.com/playframework/interplay#the-root-project) this aggregating `root` project should be a `PlayRootProject` and then, each of the aggregated projects that must not be published should be `PlayNoPublish`.

Ofc, `interplay` being created before sbt `1.x` it's possible that some assumptions no longer hold true in the sbt `1.3+` ecosystem.

Comment (ignasi35) [reply build.sbt]: I think the `NoPublish` is unnecessary and your change above:

```

  publishTo := Some(
    if (isSnapshot.value)
      Opts.resolver.sonatypeSnapshots
    else
      Opts.resolver.sonatypeStaging
  ),
```

is enough.

File: build.sbt [hunk 2]
```diff
@@ -126,7 +126,7 @@ lazy val commonSettings = Def.settings(
 
 lazy val root = project
   .in(file("."))
-  .enablePlugins(PlayRootProject, ScalaJSPlugin, PlayNoPublish)
```
Comment (marcospereira) [inline build.sbt]: As `PlayNoPublish` was removed, we need to add the settings manually. I think `publish / skip := true` should be enough (as [documented here](https://www.scala-sbt.org/1.x/docs/Publishing.html#Skip+publishing)) but maybe re-add `publishTo: None` as well.

Comment (ignasi35) [reply build.sbt]: `publish / skip := true` is what we use in other projects (eg https://github.com/playframework/play-grpc/blob/master/project/Publish.scala#L13).

I've ran `show publishTo` with this setup and the `root` project reports no target (which is what I think we wanted).

How about we merge and we revisit this on the next release?

**Pull request merged**